# Ito Integral Simulation

To demonstrate simulation of an Ito integral, we will simulate the following Ito integral 1000 times with step size 0.01...

$$\int_{0}^{2} e^{B_s}dB_s.$$

We'll then calculate the empirical variance and compare with the true result.

## Exact Solution:

To find the variance of the Ito integral...

$$ \int_{0}^{2} e^{B_s}dB_s, $$

we take...

$$ \text{Var} \left( \int_{0}^{t} e^{B_s}dB_s \right) = \mathbb{E} \left[ \left( \int_{0}^{t} e^{B_s} dB_s \right)^2 \right] - \left( \mathbb{E} \left[ \int_{0}^{t} e^{B_s} dB_s \right] \right)^2. $$

However, we know that...

$$ \mathbb{E} \left[ \int_{0}^{t} e^{B_s} dB_s \right] = 0, $$

so we are left with...

$$ \text{Var} \left( \int_{0}^{t} e^{B_s}dB_s \right) = \mathbb{E} \left[ \left( \int_{0}^{t} e^{B_s} dB_s \right)^2 \right] $$

$$ = \mathbb{E} \left[ \int_{0}^{t} e^{2B_s} dB_s \right] $$

$$ = \int_{0}^{t} \mathbb{E} \left[ e^{2B_s} \right] dB_s $$

$$ = \int_{0}^{t} e^{2s} dB_s $$

$$ = \frac{1}{2} \left(e^{2t} - 1 \right) $$

For $t=2$, we find...

$$ \text{Var} \left( \int_{0}^{2} e^{B_s}dB_s \right) =  \frac{1}{2} \left( e^{4} - 1 \right) \approx 26.7991$$

## Simulation:

We will approximate the Ito integral...

$$\int_{0}^{2} e^{B_s}dB_s \approx \sum_{k=1}^{n} e^{B_{t_{k-1}}}(B_{t_k} - B_{t_{k-1}})$$

for $t_k = 0.01k$, which gives us $n = 200$ in the interval $[0,2]$.

To simplify our simulation, we note that the increment $B_{t_k}-B_{t_{k-1}} = Z_k\sqrt{\frac{t}{n}}$ for standard Normal RVs $Z_1,Z_2,\dots,Z_n$.

In [1]:
import numpy as np
import math
import matplotlib.pyplot as plt

In [2]:
# t = end of interval in [0,t]
# n = number of increments between 0 and t
# N = number of simulations

def itoApprox(t,n):
    dt = t/n
    bIncrements = np.random.normal(0,1,n)*math.sqrt(dt)
    bPath = np.cumsum(bIncrements)
    bPath = np.insert(bPath,0,0)
    ito = 0
    for k in range(n):
        ito += math.exp(bPath[k])*bIncrements[k]
    return ito

def itoSim(N):
    np.random.seed(10)
    itoApproxs = [itoApprox(2,200) for i in range(N)]
    empVar = np.var(itoApproxs,ddof=1)
    return empVar

itoSim(1000)

25.294468688142498

Simulated above, our empirical variance of the Ito integral is about $25.2945$. We note that this value is very close to our calculated true value of $26.7991$.